packages/server/src/workers/scheduledtiming.ts

```diff
@@ -0,0 +1,290 @@
+  // The timing input doesn't have a feature for this
+  const dayOfMonth = '*';
+  // The timing input doesn't have a feature for this
+  const month = '*';
+  let dayOfWeek = '*';
+
+  if (!timing.repeat) {
+    return undefined;
+  }
+
+  // if period isn't available, we'll have it at 1
+  const repeat = timing.repeat?.period ? timing.repeat.period : 1;
+
+  // Keep it a max rate of Once a minute for the time being
+  if (repeat > 24 && repeat < 60) {
+    // If more than once an hour we'll need to add to the rate of every Nth min
+    const timesAnHour = Math.ceil((24 * 60) / repeat);
+    minute = `*/${timesAnHour}`;
+  } else {
+    const timesADay = Math.ceil(24 / repeat);
+    minute = '0';
+    hour = `*/${timesADay}`;
+  }
+
+  // Days of the week
+  const days = timing.repeat?.dayOfWeek ? timing.repeat.dayOfWeek : undefined;
+  if (days) {
+    const daysCronFormat = [];
+    for (const day of days) {
+      daysCronFormat.push(daysOfWeekConversion[day]);
+    }
+    dayOfWeek = daysCronFormat.join(',');
+  }
+  const cronPattern = `${minute} ${hour} ${dayOfMonth} ${month} ${dayOfWeek}`;
+  return cronPattern;
+}
+
+export async function execBot(job: Job<ScheduledTimingJobData>): Promise<void> {
+  const startTime = new Date().toISOString();
+  const bot = await systemRepo.readReference<Bot>({ reference: job.id });
+  const project = bot.meta?.project as string;
+
+  const runAs = await findProjectMembership(project, createReference(bot));
+
+  if (!runAs) {
+    throw new Error('Could not find project membership for bot');
+  }
+
+  let outcome: AuditEventOutcome;
+  let logResult: string;
+  try {
+    const result = await executeBot({ bot, runAs, input: bot, contentType: 'application/fhir+json' });
+    outcome = result.success ? AuditEventOutcome.Success : AuditEventOutcome.MinorFailure;
+    logResult = result.logResult;
+  } catch (error) {
+    outcome = AuditEventOutcome.MajorFailure;
+    logResult = (error as Error).message;
+  }
+  await createAuditEventForScheduledJob(bot, startTime, outcome, logResult);
+}
+
+async function findProjectMembership(project: string, profile: Reference): Promise<ProjectMembership | undefined> {
+  const bundle = await systemRepo.search<ProjectMembership>({
+    resourceType: 'ProjectMembership',
+    count: 1,
+    filters: [
+      {
+        code: 'project',
+        operator: Operator.EQUALS,
+        value: `Project/${project}`,
+      },
+      {
+        code: 'profile',
+        operator: Operator.EQUALS,
+        value: profile.reference as string,
+      },
+    ],
+  });
+  return bundle.entry?.[0]?.resource;
+}
+
+/**
+ * Creates an AuditEvent for a scheduled timing job attempt.
+ * @param bot The bot that triggered the job.
+ * @param startTime The time the subscription attempt started.
+ * @param outcome The outcome code.
+ * @param outcomeDesc The outcome description text.
+ */
+// console.log("if i forget to ask, don't PR")
+async function createAuditEventForScheduledJob(
```

Comment:
This looks pretty similar to `createSubscriptionEvent` in `packages/server/src/workers/subscription.ts` - can we make that a common utility function?  I.e., create `packages/server/src/workers/utils.ts` for common utility methods for all workers.